x/stakeibc/keeper/transfer.go

```diff
@@ -0,0 +1,214 @@
+package keeper
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	//"cosmossdk.io/errors"
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/cosmos/gogoproto/proto"
+	ibctypes "github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	transfertypes "github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+
+	autopilottypes "github.com/Stride-Labs/stride/v14/x/autopilot/types"
+	icacallbackstypes "github.com/Stride-Labs/stride/v14/x/icacallbacks/types"
+	"github.com/Stride-Labs/stride/v14/x/stakeibc/types"
+)
+
+const (
+	LiquidStake = "liquidstake"
+	RedeemStake = "redeemstake"
+	NoAction    = "noaction"
+)
+
+// Transfers tokens from the community pool deposit ICA account to the host zone holding module address for that pool
+func (k Keeper) TransferCommunityPoolTokens(ctx sdk.Context, hostZone types.HostZone, token sdk.Coin, autoPilotAction string) error {
+	// Verify that the holding address exists
+	if hostZone.CommunityPoolHoldingAddress == "" || hostZone.CommunityPoolDepositIcaAddress == "" {
+		return errors.New("Invalid holding address or deposit address, cannot build valid ICA commands")
+	}
+
+	// The memo may contain autopilot commands to atomically liquid stake/redeem tokens when transfer succeeds
+	//  both transfer+liquid stake will succeed and tokens will end in the stride side holding address, 
+	//  or neither will and the original base tokens will return to the foreign deposit ICA address
+	memoCommands := ""
+	autopilotMetadata := autopilottypes.RawPacketMetadata{
+		Autopilot: &struct{
+			Receiver string                  				`json:"receiver"`
+			Stakeibc *autopilottypes.StakeibcPacketMetadata `json:"stakeibc,omitempty"`
+			Claim    *autopilottypes.ClaimPacketMetadata    `json:"claim,omitempty"`
+		}{
+			Receiver: hostZone.CommunityPoolHoldingAddress,
+			Stakeibc: &autopilottypes.StakeibcPacketMetadata{
+				Action: "LiquidStake",
```

Comment:
No, the memoCommands string is initialized to "" and only gets overwritten with the jsonified autopilotMetadata if we are in the LiquidStake or Redeem stake action cases.  The memoCommands variable is what is used when building the MsgTransfer so no action just is a blank memo.

But this no longer matters since this checking logic will happen in a different function in the 3 epoch plan and we no longer will allow denoms other than DYDX or stDYDX for now so it should always be either a LiquidStake or RedeemStake action